### ISSUE TITLE
TypeError('integer argument expected, got int') while importing numpy-1.19.2

### ISSUE THREAD

**Issue by oroppas** [graalvm/graalpython#164]
Following the fix in [#163](https://github.com/graalvm/graalpython/issues/163), I've got the following warnings and errors:

```
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:69: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'flatiter' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core', 'flatiter', ('base',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:103: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'flatiter' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core', 'flatiter', ('index',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:677: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'broadcast' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core', 'broadcast', ('ndim',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:693: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'broadcast' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core', 'broadcast', ('nd',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:708: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'broadcast' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core', 'broadcast', ('numiter',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5236: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('alignment',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5255: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('byteorder',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5298: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('char',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5362: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('flags',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5444: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('itemsize',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5466: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('kind',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5532: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('num',
/home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/core/_add_newdocs.py:5644: UserWarning: add_newdoc was used on a pure-python object <property 'None' of 'dtype' objects>. Prefer to attach it directly to the source.
  add_newdoc('numpy.core.multiarray', 'dtype', ('type',
sys:1: RuntimeWarning: builtins.type size changed, may indicate binary incompatibility. Expected 416 from C header, got 880 from PyObject
Error in sys.excepthook:

Traceback (most recent call last):
  File "/home/ryuta/packages/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_9EE94B73E5_JAVA11/graalvm-9ee94b73e5-java11-20.3.0-dev/languages/python/lib-python/3/traceback.py", line 120, in format_exception
  File "/home/ryuta/packages/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_9EE94B73E5_JAVA11/graalvm-9ee94b73e5-java11-20.3.0-dev/languages/python/lib-python/3/traceback.py", line 481, in __init__
  File "/home/ryuta/packages/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_9EE94B73E5_JAVA11/graalvm-9ee94b73e5-java11-20.3.0-dev/languages/python/lib-python/3/traceback.py", line 509, in __init__
  File "/home/ryuta/packages/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_9EE94B73E5_JAVA11/graalvm-9ee94b73e5-java11-20.3.0-dev/languages/python/lib-python/3/traceback.py", line 360, in extract
UnboundLocalError: local variable 'name' referenced before assignment

Original exception was:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot load /home/ryuta/.local/lib/python3.8/site-packages/numpy-1.19.2-py3.8-linux-x86_64.egg/numpy/random/mtrand.graalpython-38-native-x86_64-linux.so: TypeError('integer argument expected, got int')
```

**Comment by msimacek** [comment]
Hi, @oroppas. As for this particular error, I believe it's already fixed. I was merging a fix for an error with the same symptoms on Friday and I couldn't reproduce the exception you got using your reproducer anymore. However, I ran into the same error as in #167.
As for getting numpy 1.19.2 to work, I expect there will be bigger problems than these small bugs. @fangerer, you had some details on those, could you please share them?

**Comment by fangerer** [comment]
Hi @oroppas, we do currently only support the NumPy version that is installed by our `ginstall` script because we maintain a small set of patches for NumPy. The patches are mainly about removing platform- and architecture-specific assumptions.
The exact patch is in our public repo (look for file `numpy.patch`) and can certainly not be applied to a different NumPy version.

**Comment by oroppas** [comment]
Thank you so much for your support.

@msimacek I still see the same error. Let me build everything from scratch...

@fangerer Sure. I'm aware of that. Like I said in #163, I was curious how much GraalPython can build the up-to-date numpy and hopefully import numpy.

It's very encouraging that I only needed the one minor patch to build numpy-1.19.2
I think it's as good as it can be for now.  If you don't mind, I'd like to keep this issue opened either until #167 or
the error I see fixed.

Thank you.

**Comment by msimacek** [comment]
I fixed #167 and now I can finally reproduce the issue you had, so it's not fixed yet. I'll look into it
